### ISSUE TITLE
Add support for zstd compression

### ISSUE THREAD

**Issue by travisperson** [filecoin-project/filecoin-chain-archiver#40]
zstd compression reduces the size of snapshots quite significantly we should explore how we can incorporate this compression. I have not used it before so unsure if it will work well in our current streaming model, of it this will need to be a post-processing to create a zstd compressed file and add it back.

**Comment by ribasushi** [comment]
@travisperson yup, it works with streaming. `zstd` compresses faster than lotus spits things out, it takes ~4 cores to keep up. This is **literally** what I do today:
```
lotus chain export --tipset @$height --recent-stateroots $span_epochs --skip-old-msgs \
    /dev/stdout 2> >( grep --line-buffered -v "rpc output message buffer" 1>&2 ) \
| tee >( sha256sum > "$expdir/$datedprefix.sha256sum" ) \
| tee "$expdir/$datedprefix.car" \
| zstd -T0 --long -15 > "$expdir/$datedprefix.car.zst"
```

**Comment by travisperson** [comment]
Awesome this should be really easy to add then.

**Comment by ribasushi** [comment]
This can be closed as it has been resolved by https://github.com/filecoin-project/filecoin-chain-archiver/pull/50
However see request for significant improvement at https://github.com/filecoin-project/filecoin-chain-archiver/issues/55
